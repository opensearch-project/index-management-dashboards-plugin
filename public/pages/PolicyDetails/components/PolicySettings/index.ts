/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import PolicySettings from "./PolicySettings";

export default PolicySettings;
