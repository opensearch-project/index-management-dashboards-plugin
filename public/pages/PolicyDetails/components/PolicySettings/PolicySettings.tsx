/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { Component } from "react";
import { EuiLink, EuiFlexGrid, EuiSpacer, EuiFlexItem, EuiText, EuiPanel, EuiFlexGroup, EuiHorizontalRule, EuiTitle } from "@elastic/eui";
import { ContentPanel } from "../../../../components/ContentPanel";
import { ModalConsumer } from "../../../../components/Modal";
import { ErrorNotification, ISMTemplate } from "../../../../../models/interfaces";
import JSONModal from "../../../../components/JSONModal";

interface PolicySettingsProps {
  policyId: string;
  errorNotification: ErrorNotification | null | undefined;
  primaryTerm: number;
  lastUpdated: number | undefined;
  description: string;
  sequenceNumber: number;
  ismTemplates: ISMTemplate[] | ISMTemplate | null;
}

interface PolicySettingsState {}

export default class PolicySettings extends Component<PolicySettingsProps, PolicySettingsState> {
  render() {
    const { policyId, errorNotification, primaryTerm, lastUpdated, description, sequenceNumber } = this.props;

    const updatedDate = lastUpdated ? new Date(lastUpdated).toLocaleString() : "-";

    let errorNotificationValue: string | JSX.Element = "-";

    if (errorNotification) {
      errorNotificationValue = (
        <ModalConsumer>
          {({ onShow }) => (
            <EuiLink onClick={() => onShow(JSONModal, { title: "Error notification", json: errorNotification })}>View code</EuiLink>
          )}
        </ModalConsumer>
      );
    }

    const infoItems = [
      { term: "Policy name", value: policyId },
      { term: "Error notification", value: errorNotificationValue },
      { term: "Primary term", value: primaryTerm },
      { term: "Last updated", value: updatedDate },
      { term: "Policy description", value: description || "-" },
      { term: "Sequence number", value: sequenceNumber },
    ];

    return (
      <EuiPanel>
        <EuiFlexGroup gutterSize="xs" alignItems="center">
          <EuiTitle size="s">
            <h2>{`Policy settings`}</h2>
          </EuiTitle>
        </EuiFlexGroup>
        <EuiHorizontalRule margin={"xs"} />
        <div>
          <EuiFlexGrid columns={4}>
            {infoItems.map((item) => (
              <EuiFlexItem key={`${item.term}#${item.value}`}>
                <EuiText size="s">
                  <dt>{item.term}</dt>
                  <dd>{item.value}</dd>
                </EuiText>
              </EuiFlexItem>
            ))}
          </EuiFlexGrid>
        </div>
      </EuiPanel>
    );
  }
}
