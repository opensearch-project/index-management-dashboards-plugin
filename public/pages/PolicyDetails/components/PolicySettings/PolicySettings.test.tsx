/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render } from "@testing-library/react";
import PolicySettings from "./PolicySettings";

describe("<PolicySettings /> spec", () => {
  beforeAll(() => {
    jest.useFakeTimers('modern');
    jest.setSystemTime(new Date(2021, 7, 1));
  });

  afterAll(() => {
    jest.useRealTimers();
  });

  it("renders the component", () => {
    const { container } = render(
      <PolicySettings
        policyId={"some_id"}
        channelId={"some_channel_id"}
        primaryTerm={1}
        lastUpdated={new Date().toString()}
        description={"some description"}
        sequenceNumber={2}
        schemaVersion={3}
        ismTemplates={[]}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
