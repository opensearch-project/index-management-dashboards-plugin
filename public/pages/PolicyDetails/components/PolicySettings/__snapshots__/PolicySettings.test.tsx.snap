// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PolicySettings /> spec renders the component 1`] = `
<div
  class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
  style="padding-left: 0px; padding-right: 0px;"
>
  <div
    class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsFlexStart euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
    style="padding: 0px 10px;"
  >
    <div
      class="euiFlexItem"
    >
      <h3
        class="euiTitle euiTitle--small"
      >
        Policy settings
        <span
          class="panel-header-count"
        >
           
           
        </span>
      </h3>
    </div>
  </div>
  <hr
    class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
  />
  <div
    style="padding: 10px;"
  >
    <div
      style="padding-left: 10px;"
    >
      <div
        class="euiSpacer euiSpacer--s"
      />
      <div
        class="euiFlexGrid euiFlexGrid--gutterLarge euiFlexGrid--fourths euiFlexGrid--responsive"
      >
        <div
          class="euiFlexItem"
        >
          <div
            class="euiText euiText--extraSmall"
          >
            <dt>
              Policy name
            </dt>
            <dd>
              some_id
            </dd>
          </div>
        </div>
        <div
          class="euiFlexItem"
        >
          <div
            class="euiText euiText--extraSmall"
          >
            <dt>
              Error notification
            </dt>
            <dd>
              -
            </dd>
          </div>
        </div>
        <div
          class="euiFlexItem"
        >
          <div
            class="euiText euiText--extraSmall"
          >
            <dt>
              Primary term
            </dt>
            <dd>
              1
            </dd>
          </div>
        </div>
        <div
          class="euiFlexItem"
        >
          <div
            class="euiText euiText--extraSmall"
          >
            <dt>
              Last updated
            </dt>
            <dd>
              8/1/2021, 12:00:00 AM
            </dd>
          </div>
        </div>
        <div
          class="euiFlexItem"
        >
          <div
            class="euiText euiText--extraSmall"
          >
            <dt>
              Policy description
            </dt>
            <dd>
              some description
            </dd>
          </div>
        </div>
        <div
          class="euiFlexItem"
        >
          <div
            class="euiText euiText--extraSmall"
          >
            <dt>
              Sequence number
            </dt>
            <dd>
              2
            </dd>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
